dart/lib/src/sentry_client.dart

```diff
@@ -50,6 +63,8 @@ class SentryClient {
 
   SentryStackTraceFactory get _stackTraceFactory => _options.stackTraceFactory;
 
+  final Map<Type, List<dynamic>> _lifecycleCallbacks = {};
```

Comment:
```suggestion
  final Map<Type, List<Function>> _lifecycleCallbacks = {};
```

I think we can avoid dynamic